cmd/thanos/sidecar.go

```diff
@@ -298,13 +338,31 @@ func runSidecar(
 	return nil
 }
 
+func validatePrometheus(ctx context.Context, logger log.Logger, promURL *url.URL, tsdbPath string) error {
+	flags, err := promclient.ConfiguredFlags(ctx, logger, promURL)
+	if err != nil {
+		return errors.Wrap(err, "configured flags; failed to check flags")
```

Comment:
minor nit: just write `failed to check flags` here IMO, you already say above that we are talking about validating the configured flags